code/game/machinery/pipe/construction.dm

```diff
@@ -159,33 +159,26 @@ Buildable meters
 	var/flags = initial(fakeA.pipe_flags)
 	var/pipe_count = 0
 	for(var/obj/machinery/atmospherics/machine in loc)
-		if(machine.piping_layer != piping_layer)
+		if(machine.piping_layer != piping_layer && !((machine.pipe_flags | flags) & PIPING_ALL_LAYER))
 			continue
 		pipe_count += 1
 	for(var/obj/machinery/atmospherics/machine in loc)
 		if((machine.pipe_flags & flags & PIPING_ONE_PER_TURF)) //Only one dense/requires density object per tile, eg connectors/cryo/heater/coolers.
 			to_chat(user, span_warning("Something is hogging the tile!"))
 			return TRUE
```

Comment:
Could this be done in the other for loop?

Reply:
Done